### PR TITLE
[12.0][FIX] fix, an unactive variant should stay inactive

### PR DESCRIPTION
**Pull request by sebastienbeau**
@kevinkhao @mourad-ehm @bealdav This may interest you

### CODE REVIEW

**Review by kevinkhao — APPROVED** [review]

File: product_variant_inactive/tests/test_button.py [hunk 1]
```diff
@@ -58,3 +58,11 @@ def help_create_product(self, active=True):
             {"active": active, "name": "test_product"}
         )
         return product
+
+    def test_create_variant_do_not_reactive(self):
```
Comment (kevinkhao) [inline product_variant_inactive/tests/test_button.py]: suggest: test_create_variant_do_not_reactive -> test_create_variant_do_not_reactivate

File: product_variant_inactive/tests/test_button.py [hunk 2]
```diff
@@ -58,3 +58,11 @@ def help_create_product(self, active=True):
...
+        # Ensure that generating againt the variant do not change
```
Comment (kevinkhao) [inline product_variant_inactive/tests/test_button.py]: suggest: Ensure that re-generating variants does not change the "active" state of the existing variant

File: product_variant_inactive/models/product_variant.py [hunk 3]
```diff
@@ -37,3 +40,11 @@ def fields_view_get(
                     orm.setup_modifiers(button, root)
             res["arch"] = etree.tostring(root, pretty_print=True)
         return res
+
+    @api.multi
+    def write(self, vals):
+        if self._context.get("no_reactive") and vals == {"active": True}:
```
Comment (kevinkhao) [inline product_variant_inactive/models/product_variant.py]: suggest: no_reactive -> no_reactivate or no_reactivation

**Review by kevinkhao — COMMENTED** [review]
nitpicking

File: product_variant_inactive/models/product_variant.py (hunk 3 above)
Comment (sebastienbeau) [reply product_variant_inactive/models/product_variant.py]: fixed

File: product_variant_inactive/tests/test_button.py (hunk 2 above)
Comment (sebastienbeau) [reply product_variant_inactive/tests/test_button.py]: fixed

File: product_variant_inactive/tests/test_button.py (hunk 1 above)
Comment (sebastienbeau) [reply product_variant_inactive/tests/test_button.py]: fixed

**Review by HaraldPanten — APPROVED** [review]

**Review by bealdav — APPROVED** [review]

**Pull request merged**